apps/web/test/CodeEditor.test.tsx

```
@@ -0,0 +1,177 @@
+    <textarea
+      data-testid="base-code-editor"
+      value={value}
+      onChange={(e) => onChange(e.target.value)}
+    />
+  ),
+}));
+
+vi.mock('@/components/Battle/Player/EditorFooter', () => ({
+  default: ({ onDryRun, onSubmit }: { onDryRun: () => void; onSubmit: () => void }) => (
+    <div data-testid="editor-footer">
+      <button onClick={onDryRun}>Run</button>
```

Biome (2.4.4)

[error] Provide an explicit type prop for the button element.

(lint/a11y/useButtonType)